File: tests/fixtures/rust/crypto_usage.rs
// Fixture: Rust crypto usage across the risk spectrum.
//
// Known answers (all medium confidence; crate/token names intentionally not
// spelled out in this header so only the real code lines match):
// - the md5 crate use            -> MD5, P0 classically broken
// - the openssl aes-128-ecb call -> AES-128 + ECB, P0 broken usage
// - the openssl RSA keygen       -> RSA 2048, P0 quantum-broken + HNDL
// - the aes-gcm crate use        -> AES + GCM (key size unknown), P2
// - the chacha20poly1305 use     -> CHACHA20, compliant

use md5::{Digest, Md5}; // KNOWN: MD5, P0
use aes_gcm::Aes256Gcm; // KNOWN: AES + GCM (size not inferred from crate), P2
use chacha20poly1305::ChaCha20Poly1305; // KNOWN: CHACHA20, compliant
use openssl::symm::Cipher;
use openssl::rsa::Rsa;

fn weak_checksum(data: &[u8]) -> Vec<u8> {
    let mut hasher = Md5::new();
    hasher.update(data);
    hasher.finalize().to_vec()
}

fn legacy_cipher() -> Cipher {
    Cipher::aes_128_ecb() // KNOWN: AES-128 + ECB, P0
}

fn quantum_vulnerable() -> Rsa<openssl::pkey::Private> {
    Rsa::generate(2048).unwrap() // KNOWN: RSA 2048, P0
}

File: tests/fixtures/ruby/crypto_usage.rb
# Fixture: Ruby crypto across the risk spectrum.
#
# Known answers (all medium confidence; identifiers below are the real call
# sites — the header intentionally avoids matchable tokens):
# - the MD5 digest              -> MD5, P0 classically broken
# - the aes-128-ecb cipher      -> AES-128 + ECB, P0 broken usage
# - the RSA key (2048)          -> RSA 2048, P0 quantum-broken + HNDL
# - the aes-256-gcm cipher      -> AES-256 + GCM, compliant

require 'openssl'

def weak_digest(data)
  OpenSSL::Digest::MD5.new.digest(data)          # KNOWN: MD5, P0
end

def legacy_cipher
  OpenSSL::Cipher.new('aes-128-ecb')             # KNOWN: AES-128 + ECB, P0
end

def quantum_vulnerable
  OpenSSL::PKey::RSA.new(2048)                    # KNOWN: RSA 2048, P0
end

def modern_seal
  OpenSSL::Cipher.new('aes-256-gcm')             # KNOWN: AES-256 + GCM, compliant
end
